prov/tcp/src/tcpx_ep.c

```diff
@@ -349,23 +330,30 @@ int tcpx_endpoint(struct fid_domain *domain, struct fi_info *info,
 	ret = ofi_endpoint_init(domain, &tcpx_util_prov, info, &ep->util_ep,
 				context, tcpx_manual_progress);
 	if (ret)
-		goto err;
+		goto err1;
 
-	ep->info = *info;
-	if (info && info->handle) {
-		handle = container_of(info->handle,
-				      struct tcpx_conn_handle,
+	if (info->handle) {
+		handle = container_of(info->handle, struct tcpx_conn_handle,
 				      handle);
 		ep->conn_fd = handle->conn_fd;
 		free(handle);
 	} else {
-		ep->conn_fd = ofi_socket(AF_INET, SOCK_STREAM, 0);
+		if (info->src_addr)
+			af = ((const struct sockaddr *) info->src_addr)->sa_family;
+		else if (info->dest_addr)
+			af = ((const struct sockaddr *) info->dest_addr)->sa_family;
+		else
+			af = ofi_get_sa_family(info->addr_format);
+
+		ep->conn_fd = ofi_socket(af, SOCK_STREAM, 0);
 		if (!ep->conn_fd) {
-		  ret = -errno;
-		  goto err;
+			ret = -errno;
+			goto err1;
```

Comment:
err2

Reply:
Actually, err1 is the correct location to jump to.  Though this pointed out that the check was incorrect.  It should have been conn_fd == INVALID_SOCKET, rather than !conn_fd.